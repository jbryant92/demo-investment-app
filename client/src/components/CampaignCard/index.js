import React, { Component } from 'react';
import {
  Card,
  Image,
  Icon,
  Progress
} from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types'

class CampaignCard extends Component {
  render() {
    const {
      id,
      imageUrl,
      name,
      sector,
      country,
      targetAmount,
      fundedAmount,
      percentageComplete,
      numberOfInvestors
    } = this.props.campaign

    const clickable = this.props.clickable
    const inverstorString = numberOfInvestors > 1 ? 'people have' : 'person has'

    return (
      <Card key={ id } as={ clickable ? Link : null } to={ clickable ? `/campaigns/${id}` : null }>
        <Image src={ imageUrl } />

        <Card.Content>
          <Card.Header>{ name }</Card.Header>
          <Card.Meta>{ sector }</Card.Meta>
          <Card.Meta>{ country.name }</Card.Meta>
          <Card.Description>Investment sought: £{ targetAmount }</Card.Description>
          <Card.Description>Investment already funded: £{ fundedAmount }</Card.Description>
        </Card.Content>

        <Card.Content extra>
          <Progress
            percent={ percentageComplete }
            size='small'
            active
            success={ percentageComplete >= 100.0 }
            color='blue'
          >
            { percentageComplete }% complete
          </Progress>
        </Card.Content>

        <Card.Content extra>
          <Icon name='user' />
          { numberOfInvestors } { inverstorString } invested
        </Card.Content>
      </Card>
    )
  }
}

CampaignCard.propTypes = {
  campaign: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    sector: PropTypes.string.isRequired,
    targetAmount: PropTypes.number.isRequired,
    fundedAmount: PropTypes.number.isRequired,
    percentageComplete: PropTypes.number.isRequired,
    numberOfInvestors: PropTypes.number.isRequired,
    country: PropTypes.shape({
      name: PropTypes.string
    }).isRequired
  })
}

export default CampaignCard;
