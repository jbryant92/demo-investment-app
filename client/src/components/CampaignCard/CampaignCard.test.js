import React from 'react';
import ReactDOM from 'react-dom';
import CampaignCard from './index';
import enzymeToJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const mockCampaign = {
    id: 1,
    name: 'foo',
    sector: 'bar',
    targetAmount: 100,
    fundedAmount: 50,
    percentageComplete: 50,
    numberOfInvestors: 2,
    country: {
      name: 'uk'
    }
  }
  const instance = shallow(
        <CampaignCard campaign={ mockCampaign } />
    )
    expect(enzymeToJson(instance)).toMatchSnapshot()
});
