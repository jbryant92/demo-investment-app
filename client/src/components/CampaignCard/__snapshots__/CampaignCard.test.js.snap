// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Card
  as={null}
  key="1"
  to={null}
>
  <Image
    as="img"
    ui={true}
  />
  <CardContent>
    <CardHeader>
      foo
    </CardHeader>
    <CardMeta>
      bar
    </CardMeta>
    <CardMeta>
      uk
    </CardMeta>
    <CardDescription>
      Investment sought: £
      100
    </CardDescription>
    <CardDescription>
      Investment already funded: £
      50
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <Progress
      active={true}
      color="blue"
      percent={50}
      size="small"
      success={false}
    >
      50
      % complete
    </Progress>
  </CardContent>
  <CardContent
    extra={true}
  >
    <Icon
      as="i"
      name="user"
    />
    2
     
    people have
     invested
  </CardContent>
</Card>
`;
